shells/pipes-shell/pipe.js

```diff
@@ -52,38 +48,38 @@ const extractType = json => {
   return (entity ? entity.type : 'com.music.spotify').replace(/\./g, '_');
 };
 
+const pipeHandlers = {};
+
 const dispatch = async (type, arc, callback) => {
-  await instantiatePipeRecipe(arc, type);
-  switch (type) {
-    case 'com_google_android_apps_maps':
-      await com_google_android_apps_maps(arc, callback);
-      break;
-    default:
-    case 'com_music_spotify':
-      await com_music_spotify(arc, callback);
-      break;
+  const handler = pipeHandlers[type];
+  if (handler) {
+    await instantiatePipeRecipe(arc, type);
+    await handler(arc, callback);
   }
 };
 
```

Comment:
These two methods look roughly the same except for the manifest name, and instantiateRecipe's third argument.  Seems like it could be simplified into one method and a simple json configuration, e.g. 

handlerDefinitions: [{com_music_spotify: { manifest: 'http://...', name: "ArtistAutofill"}, ...]

And then I have to wonder if these handler definitions can't live in a master file somewhere?